qdev_wrappers/qdev_fitter.py

```diff
@@ -0,0 +1,141 @@
+import qcodes as qc
+import numpy as np
+import types
+from os.path import sep
+import matplotlib.pyplot as plt
+import scipy.fftpack as fftpack
+from scipy.optimize import curve_fit
+
+from qdev_wrappers.file_setup import CURRENT_EXPERIMENT
+
+
+
+
+def qdev_fitter(id, fitfunction_in, do_plots=True, dataname=None, samplefolder=None, p0=None,**kwargs):
+    '''
+    Function to allow easy on the fly fitting of data. Only works for 1D datasets at the moment.
+    id (num): Id of dataset
+    fitfunction_in (str, function): Function data will be fitted to. Can be string for functions defined here or a function passed direclty defined elsewhere.
+    do_plots (bool): If plots are done. Will be saved in the analysis folder for the samplepath
+    dataname (str): String with dataname in the qcodes dataset
+    samplefolder (path): Path to samplefolder if not opening active measurements
+    p0 (tuple): Initial parameters for fit
+    kwargs: Passed to scipy.optimize.curve_fit when fitting.
+    '''
+
+    # Load data from active samplefolder or input samplefolder.
```

Comment:
Ok, I'll definitely refactor this out, as the next lines get used in three different places now...

Reply:
This is the exact same code as in show_num and show_meta. Refactoring to somewhere else is fine for me. Not sure where it should go though. Maybe just in the show_num file?